overlord/snapstate/component.go

```diff
@@ -96,7 +96,23 @@ type componentInstallTaskSet struct {
 	beforeLink         []*state.Task
 	linkToHook         []*state.Task
 	postOpHookAndAfter []*state.Task
-	ts                 *state.TaskSet
+}
+
+func (c *componentInstallTaskSet) taskSet() (*state.TaskSet, error) {
+	tasks := make([]*state.Task, 0, len(c.beforeLink)+len(c.linkToHook)+len(c.postOpHookAndAfter))
+	tasks = append(tasks, c.beforeLink...)
+	tasks = append(tasks, c.linkToHook...)
+	tasks = append(tasks, c.postOpHookAndAfter...)
+
+	if len(c.linkToHook) == 0 {
+		return nil, errors.New("internal error: link component task set does not have a link task")
+	}
+
+	ts := state.NewTaskSet(tasks...)
+	ts.MarkEdge(c.compSetupTask, BeginEdge)
+	ts.MarkEdge(c.linkToHook[0], MaybeRebootEdge)
```

Comment:
Thinking about it, probably we don't want to reboot when kernel-modules components are installed and maybe this edge should be removed for the moment.

Reply:
Edge is removed now.